commands/cli.go

```diff
@@ -0,0 +1,290 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/urfave/cli"
+	"github.com/vapor-ware/sctl/cloud"
+	"github.com/vapor-ware/sctl/utils"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+	"time"
+)
+
+// BuildContextualMenu - Assemble the CLI commands, subcommands, and flags
+// Handles the majority of the CLI interface.
+// Retuns an array of cli.Command configuration
+func BuildContextualMenu() []cli.Command {
+	return []cli.Command{
+		{
+			Name:  "add",
+			Usage: "add secret",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:   "key",
+					EnvVar: "SCTL_KEY",
+					Usage:  "GCloud KMS Key URI",
+				},
+				cli.BoolFlag{
+					Name:  "no-decode",
+					Usage: "When reading the secret, do not base64 decode",
+				},
+			},
+			Action: func(c *cli.Context) error {
+
+				// Check for KMS key uri, and presence of the secrets name
+				err := validateContext(c, "add")
+				if err != nil {
+					log.Fatal(err)
+				}
+
+				var plaintext []byte
+
+				// Scan for data being piped via STDIN and favor this over alternate inputs
+				plaintext = stdinScan()
+				if plaintext == nil {
+					// NO data deteced on stdin, attempt to scan for args after keyname
```

Comment:
Is the 'keyname' referenced here the same as the `secretName` that gets set below? if so, maybe change this to 'secret name'?